pimd/pim6_cmd.c

```diff
@@ -448,6 +451,424 @@ DEFPY (no_ipv6_pim_rp_prefix_list,
+		return CMD_WARNING;
+
+	ip_pim_ssm_show_group_range(v->info, vty, uj);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (show_ipv6_pim_interface,
+       show_ipv6_pim_interface_cmd,
+       "show ipv6 pim [mlag] [vrf NAME] interface [detail$detail|WORD$word] [json$json]",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       "MLAG\n"
+       VRF_CMD_HELP_STR
+       "PIM interface information\n"
+       "Detailed output\n"
+       "interface name\n"
+       JSON_STR)
+{
+	struct vrf *v;
+	bool uj = !!json;
+	json_object *json_parent = NULL;
+
+	v = vrf_lookup_by_name(vrf ? vrf : VRF_DEFAULT_NAME);
+
+	if (!v)
+		return CMD_WARNING;
+
+	if (uj)
+		json_parent = json_object_new_object();
+
+	if (detail)
+		pim_show_interfaces_single(v->info, vty, detail, false,
+					   json_parent);
+	else if (word)
+		pim_show_interfaces_single(v->info, vty, word, false,
+					   json_parent);
+	else
+		pim_show_interfaces(v->info, vty, false, json_parent);
+
+	if (uj)
+		vty_json(vty, json_parent);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (show_ipv6_pim_interface_vrf_all,
+       show_ipv6_pim_interface_vrf_all_cmd,
+       "show ipv6 pim [mlag] vrf all interface [detail$detail|WORD$word] [json$json]",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       "MLAG\n"
+       VRF_CMD_HELP_STR
+       "PIM interface information\n"
+       "Detailed output\n"
+       "interface name\n"
+       JSON_STR)
+{
+	bool uj = !!json;
+	struct vrf *v;
+	json_object *json_parent = NULL;
+	json_object *json_vrf = NULL;
+
+	if (uj)
+		json_parent = json_object_new_object();
+
+	RB_FOREACH (v, vrf_name_head, &vrfs_by_name) {
+		if (!uj)
+			vty_out(vty, "VRF: %s\n", v->name);
+		else
+			json_vrf = json_object_new_object();
+		if (detail)
+			pim_show_interfaces_single(v->info, vty, detail, false,
+						   json_vrf);
+		else if (word)
+			pim_show_interfaces_single(v->info, vty, word, false,
+						   json_vrf);
+		else
+			pim_show_interfaces(v->info, vty, false, json_vrf);
+
+		if (uj)
+			json_object_object_add(json_parent, v->name, json_vrf);
+	}
+	if (uj)
+		vty_json(vty, json_parent);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (show_ipv6_pim_join,
+       show_ipv6_pim_join_cmd,
+       "show ipv6 pim [vrf NAME] join [X:X::X:X$s_or_g [X:X::X:X$g]] [json$json]",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       VRF_CMD_HELP_STR
+       "PIM interface join information\n"
+       "The Source or Group\n"
+       "The Group\n"
+       JSON_STR)
+{
+	pim_sgaddr sg = {0};
+	struct vrf *v;
+	struct pim_instance *pim;
+	json_object *json_parent = NULL;
+
+	v = vrf_lookup_by_name(vrf ? vrf : VRF_DEFAULT_NAME);
+
+	if (!v) {
+		vty_out(vty, "%% Vrf specified: %s does not exist\n", vrf);
+		return CMD_WARNING;
+	}
+	pim = pim_get_pim_instance(v->vrf_id);
+
+	if (!pim) {
+		vty_out(vty, "%% Unable to find pim instance\n");
+		return CMD_WARNING;
+	}
+
+	if (!IPV6_ADDR_SAME(&s_or_g, &in6addr_any)) {
+		if (!IPV6_ADDR_SAME(&g, &in6addr_any)) {
+			memcpy(&sg.src, &s_or_g, sizeof(s_or_g));
```

Comment:
`sg.src = s_or_g;` please (and also below)

(`memcpy` has a lot fewer safety checks against accidents & mistakes.)

Reply:
Resolved, Thank you.